language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolService.java

```diff
@@ -137,6 +141,89 @@ public CompletableFuture<ExpressionTypeResponse> type(ExpressionTypeRequest requ
         });
     }
 
+    @JsonRequest
+    public CompletableFuture<TypesFromExpressionResponse> getTypeFromExpression(TypeFromExpressionRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            TypesFromExpressionResponse typesResponse = new TypesFromExpressionResponse();
+            String fileUri = request.getDocumentIdentifier().getUri();
+            String[] pathSegments = fileUri.split("/");
+            String fileName = pathSegments[pathSegments.length - 1];
+            Optional<Path> filePath = PathUtil.getPathFromURI(fileUri);
+            if (filePath.isEmpty()) {
+                return typesResponse;
+            }
+            List<ResolvedTypeForExpression> types = new ArrayList<>();
+            for (ExpressionRange range: request.getExpressionRanges()) {
+                ResolvedTypeForExpression resolvedType = new ResolvedTypeForExpression(range);
+                try {
+                    Optional<SemanticModel> semanticModel =
+                            this.workspaceManagerProxy.get().semanticModel(filePath.get());
```

Comment:
If this method is being called from the diagram editor, we should pass the `fileUri` to the get() method here and get the cloned workspace manager dedicated for the diagram editor/expression editor.

Reply:
Fixed with f8d2f6eee7fb44e6584f4520de64d82fd3d4533c